[KEYCLOAK-11436] Update JavaScript adapter documentation to prefer native promises

jonkoops opened the pull request.

jonkoops commented:
Since native Promises are not supported by all platforms (such as Internet Explorer). It would make sense to include a section where the user is informed of which polyfills may have to be included.

However I cannot find any information about which browsers are supported by Keycloak in the documentation.

jonkoops committed:
[KEYCLOAK-11436] Update JavaScript adapter documentation to prefer native promises

jonkoops commented:
No idea why the external link check is failing, seems to be a fluke.

andymunro commented on securing_apps/topics/oidc/javascript-adapter.adoc:
```diff
@@ -254,6 +270,14 @@ Example polyfill libraries:
 
+==== A note about `promiseType`
+
+Throughout the examples in the documentation you will find `promiseType: 'native'` specified wherever Keycloak is initialized.
```
Should Keycloak be {project_name} throughout this content?
Otherwise, looks good.

jonkoops replied:
Good catch, I've added it in the places where it seemed appropriate. It also seems like there are a lot of instances where this is not consistently adhered to as well, but that is outside of the scope of this PR.

andymunro reviewed (dismissed):
Thanks. Looks good.

jonkoops commented:
@stianst should we go ahead and merge this into the documentation?

jonkoops commented:
I've updated the code with an example of how to get the right API for native promises together with TypeScript. I've also replaced all instances where the function constructor of Keycloak was used with a preference towards using `new Keycloak()` to make the documentation more consistent.

jonkoops commented:
Looks like `[source,typescript]` is not supported, changed it back to `[source,javascript]`.

jonkoops commented:
@andymunro Could I ask you for one more review?
